spec/Dockerfile

```diff
@@ -5,6 +5,9 @@ FROM puppet/puppetserver
 
 COPY ./fixtures/conf/auth.conf /etc/puppetlabs/puppetserver/conf.d/
 
+RUN cd /opt/puppetlabs/puppet/modules && \
+    git clone https://github.com/puppetlabs/puppetlabs-test_device.git test_device
```

Comment:
you fix that in the "add spinner" commit, with a bit more rebase juggling, you can get rid of the double modification and clean up history.